yunwoo/programmers-2/괄호-회전하기.js

```diff
@@ -0,0 +1,86 @@
+function isCheckBracket(arr) {
+  const stack = [];
+  const brackets = {
+    '}': '{',
+    ')': '(',
+    ']': '[',
+  };
+
+  for (const bracket of arr) {
+    if (bracket === '[' || bracket === '{' || bracket === '(') {
+      stack.push(bracket);
+    } else {
+      const lastStack = stack.pop();
+
+      if (brackets[bracket] !== lastStack) {
+        return false;
+      }
+    }
+  }
+
+  return stack.length ? false : true;
+}
+
+function solution(s) {
+  const queue = s.split('');
+  let result = 0;
+
+  for (let i = 0; i < s.length; i++) {
+    if (isCheckBracket(queue)) result++;
+
+    const dequeue = queue.shift();
+
+    queue.push(dequeue);
+  }
+
+  return result;
+}
```

Comment:
```suggestion
function isCheckBracket(arr) {
  const stack = [];
  const bracketMappingObj = {
    '}': '{',
    ')': '(',
    ']': '[',
  };

  for (const bracket of arr) {
    if (!bracketMappingObj[bracket]) {
      stack.push(bracket);
    } else {
      const lastStack = stack.pop();

      if (bracketMappingObj[bracket] !== lastStack) {
        return false;
      }
    }
  }

  return stack.length ? false : true;
}

function solution(s) {
  const brackets = s.split('');
  let result = 0;

  for (let i = 0; i < s.length; i++) {
    if (isCheckBracket(brackets)) result++;

    const firstBracket = brackets.shift();

    brackets.push(firstBracket);
  }

  return result;
}
```
- 풀이잘봤습니다 🙇🏻‍♂️
- dequeue는 queue의 첫 번째 요소를 의미하는게 아니고  "double-ended queue"의 약자로, 양쪽 끝에서 삽입과 삭제가 가능한 특이한 큐 자료구조를 의미합니다. 즉 여기서 deque는 의미가 맞지 않는 네이밍인 것 같습니다.
- 그리고 s의 배열이 queue의 형태를 띄기때문에 queue라고 하신 것 같은데 기존의 dequeue도 그렇고 좀 더 자연스러운 네이밍을 위해 brackets로 변경하였습니다.
- 기존의 brackets는 mapping이라는 목적의 객체기 때문에 bracketMappingObj로 변경하였습니다.
- || 연산자로 체크하는 부분도 그냥 bracketMappingObj에 키로 접근했을 때 value의 존재 유/무로 판단이 가능하기 때문에 수정하였습니다 ~
- 괄호 문제에서 객체를 활용하는건 제가 이전에도 보여드렸었쥬 잘하셨습니다 최고최고 👍

Reply:
- 아하.. 뺀다에 dequeue 삽입한다에 inqueue인줄 알았는데 다른뜻이였군요..!! 네이밍 조금 더 체크해보겠습니다!!
- 와 ||연산자가 필요없었네요!! 👍
- 맞습니다 괄호문제!! 바로 생각하진 못했지만요😂 코멘트 감사합니다🙇‍♂️